integration-test/src/main/java/org/apache/iotdb/itbase/runtime/SortedClusterTestResultSet.java

```diff
@@ -0,0 +1,1218 @@
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.net.URL;
+import java.sql.Array;
+import java.sql.Blob;
+import java.sql.Clob;
+import java.sql.Date;
+import java.sql.NClob;
+import java.sql.Ref;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.RowId;
+import java.sql.SQLException;
+import java.sql.SQLWarning;
+import java.sql.SQLXML;
+import java.sql.Statement;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class SortedClusterTestResultSet implements ResultSet {
+
+  private final List<ResultSet> resultSets;
+  private final List<String> endpoints;
+  private final int queryTimeoutSeconds;
+  private final List<List<StringBuilder>> cachedResult = new ArrayList<>();
+
+  public SortedClusterTestResultSet(
+      List<Statement> statements, List<String> endpoints, String sql, int queryTimeoutSeconds)
+      throws SQLException {
+    this.queryTimeoutSeconds = queryTimeoutSeconds;
+    this.endpoints = endpoints;
+    RequestDelegate<ResultSet> delegate = createRequestDelegate();
+    for (Statement st : statements) {
+      delegate.addRequest(() -> st.executeQuery(sql));
+      cachedResult.add(new ArrayList<>());
+    }
+    resultSets = delegate.requestAll();
+  }
+
+  @Override
+  public boolean next() throws SQLException {
+    RequestDelegate<Boolean> delegate = createRequestDelegate();
+    for (ResultSet rs : resultSets) {
+      delegate.addRequest(rs::next);
+    }
+
+    boolean result = delegate.requestAllAndCompare();
+    if (result) {
+      for (List<StringBuilder> list : cachedResult) {
+        list.add(new StringBuilder());
+      }
+    }
+    return result;
+  }
+
+  @Override
+  public void close() throws SQLException {
+    RequestDelegate<Void> delegate = createRequestDelegate();
+    for (ResultSet rs : resultSets) {
+      delegate.addRequest(
+          () -> {
+            rs.close();
+            return null;
+          });
+    }
+    delegate.requestAll();
+
+    List<String> result = new ArrayList<>();
+    for (StringBuilder item : cachedResult.get(0)) {
+      result.add(item.toString());
+      Collections.sort(result);
+    }
+
+    for (int i = 1; i < cachedResult.size(); i++) {
+      List<String> anotherResult = new ArrayList<>();
+
+      for (StringBuilder item : cachedResult.get(i)) {
+        anotherResult.add(item.toString());
+      }
+
+      Collections.sort(anotherResult);
+
+      if (!result.equals(anotherResult)) {
+        throw new InconsistentDataException(cachedResult, endpoints);
+      }
+    }
+  }
+
+  private <T> T cachedResultAndReturnOne(List<T> data) {
+    for (int i = 0; i < data.size(); i++) {
+      List<StringBuilder> list = cachedResult.get(i);
+      list.get(list.size() - 1).append(data.get(i).toString()).append(",");
```

Comment:
If `data.get(i)` returns `null`, there will be an NullPointerException thrown. I think it's better to use `List<Set<List<Object>>>` to cache the result data, which is designed as the follow:
1. The first dimension is the node, which is the same as the current design.
2. The second dimension is the row results of one node, as the row sequence is unpredictable, we use a set to do later   comparsion.
3. The thrid dimension is the column result of one row, as we may have different data types in one row, we use a generic list to save them.

Then we can simply use `Objects.equal(cachedResult.get(0), cachedResult.get(i))` to check whether they are consistent.

Reply:
If you want the error message to be more precisemore, You can use `List<Set<Map<String, Object>>>` to save the cache.
The third dimension is the map from 'columnName' to 'columnValue'.

Reply:
Yeah, using `Set` to do the consistency check is simple, but using `Set` can't find the last record item, which used by getXXX() method like `getString(String columnLabel)` or `getString(int columnIndex)` in this ResultSet class. 

The detail is that the record item is one line of data that includes all information about one timeseries, we use the `next()` method in this ResultSet class to change to a new record item to append the information of the next timeseries, if we don't invoke the `next()` method, all getXXX() method will append to the last record item in `cachedResult`, i.e. the sequence is a big key point.

Reply:
OK, maybe we can use `List<List<Map<String, Object>>>` in the future, store columnName to columnValue in each map,  and convert to a list in the final comparsion.